script/hassfest/quality_scale_validation/runtime_data.py

```diff
@@ -0,0 +1,41 @@
+"""Enforce that the integration uses ConfigEntry.runtime_data to store runtime data.
+
+https://developers.home-assistant.io/docs/core/integration-quality-scale/rules/runtime-data
+"""
+
+import ast
+
+from script.hassfest.model import Integration
+
+
+def _has_attribute_access(tree: ast.AST, node_name: str, attribute_name: str) -> bool:
+    """Check if a node with the specified attribute is accessed."""
```

Comment:
What does "is accessed" mean?
Can we check for `writes field` in ast?

Reply:
From https://docs.python.org/3/library/ast.html#ast.Attribute it can mean a `Load`, `Store`, or `Del` so we can explicitly check it is a `Store` here.

Reply:
And can you return the type hint instead of boolean?

Then in a follow-up we can check that common methods (eg. unload/migrate) use the same type hint for the corresponding parameter.

Reply:
Where you looking for node.ctx to be returned here? I may not quite yet understand how these will be used in unload/migrate, but those seem like function calls and not an attribute access.